src/kodeagent/pattern_detector.py

```
@@ -0,0 +1,245 @@
+    """
+    try:
+        tree = ast.parse(code)
+    except SyntaxError as e:
+        return False, f'Syntax error: {str(e)}', 100
+    
+    detector = SecurityPatternDetector()
+    detector.visit(tree)
+    
+    # Determine if code is safe based on violations
+    critical_violations = [v for v in detector.violations if v[0] == 'CRITICAL']
+    high_violations = [v for v in detector.violations if v[0] == 'HIGH']
```

Ruff (0.14.7)

Local variable `high_violations` is assigned to but never used

Remove assignment to unused variable `high_violations`

(F841)